packages/picasso.js/src/core/chart-components/labels/strategies/bars.js

```diff
@@ -167,10 +167,10 @@ export function findBestPlacement({
     largest = !p || testBounds.height > largest.height ? testBounds : largest;
 
     if (orientation === 'v' && ((fitsHorizontally && testBounds.height >= measured.height)
-      || (!fitsHorizontally && testBounds.height >= measured.width && testBounds.width >= measured.height))) {
+      || (!fitsHorizontally && testBounds.height >= measured.width && (!!placement.overflow || testBounds.width >= measured.height)))) {
```

Comment:
Let's not write nested if statements like this. Break each condition down into its own statement similar to:

```js
if (orientation === 'v') {
      if (fitsHorizontally) {
        if (testBounds.height >= measured.height) {
          bounds = testBounds;
          break;
        }
      } else if (testBounds.height >= measured.width && testBounds.width >= measured.height) {
        bounds = testBounds;
        break;
      }
    }
```

Or even better. Find a way to reduce the complexity of this function!

Reply:
Refactored!